golem-worker-service-base/src/service/api_definition.rs

```diff
@@ -112,14 +112,14 @@ pub enum ApiRegistrationError {
 pub struct RegisterApiDefinitionDefault<Namespace, AuthCtx> {
     pub auth_service: Arc<dyn AuthService<AuthCtx, Namespace> + Sync + Send>,
     pub register_repo: Arc<dyn ApiDefinitionRepo<Namespace> + Sync + Send>,
-    pub api_definition_validator: Arc<dyn ApiDefinitionValidatorService + Sync + Send>,
+    pub api_definition_validator: Arc<dyn ApiDefinitionValidatorService<AuthCtx> + Sync + Send>,
```

Comment:
@nicoburniske The type annotation conceptually says, validation requires some operation that requires authorisation. This might be due to the fact we authorise getting the template.

L:et me know if I am wrong here. I think, separation of concerns is the problem

I think Validation can be just a mere function, and let the top level worker-server service layer be responsible for fetching the template metadata, and then call this reusable validation function. This way validation can exist as a pure function testable without worrying about permissions.

 Or if we still think validation itself should be trait interface, still, worker-service (the caller) should be responsible for fetching the template metadata from an outside service, such that validation can stay as pure and testable as possible

Reply:
This makes perfect sense. Much better approach